packages/features/bookings/Booker/components/DatePicker.tsx

```diff
@@ -35,6 +36,21 @@ export const DatePicker = ({
     shallow
   );
   const nonEmptyScheduleDays = useNonEmptyScheduleDays(schedule?.data?.slots);
+  const browsingDate = month ? dayjs(month) : dayjs().startOf("month");
+
+  const onMonthChange = (date: Dayjs) => {
+    setMonth(date.format("YYYY-MM"));
+    setSelectedDate(date.format("YYYY-MM-DD"));
+    setDayCount(null); // Whenever the month is changed, we nullify getting X days
+  };
+
+  useEffect(() => {
+    const currentMonth = dayjs().startOf("month").format("YYYY-MM");
+    const browsingMonth = browsingDate.format("YYYY-MM");
+    if (currentMonth === browsingMonth && nonEmptyScheduleDays.length === 0) {
+      onMonthChange(browsingDate.add(+1, "month"));
+    }
+  }, []);
```

Comment:
I think we might not need useEffect here. There is an end condition in the code that would just not call `onMonthChange`. So, technically, simply moving the code out of useEffect should work.
https://react.dev/learn/you-might-not-need-an-effect


Also, let's refactor it out in a function like `moveToNextMonthOnNoAvailability`.

Also, it would be good if we can introduce a flag to toggle this feature on/off, the default could be on though.